tests/components/comelit/test_diagnostics.py

```diff
@@ -0,0 +1,80 @@
+"""Tests for Comelit Simplehome diagnostics platform."""
+
+from __future__ import annotations
+
+from unittest.mock import patch
+
+from syrupy import SnapshotAssertion
+
+from homeassistant.components.comelit.const import DOMAIN
+from homeassistant.components.comelit.diagnostics import TO_REDACT
+from homeassistant.components.diagnostics import REDACTED
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    BRIDGE_DEVICE_QUERY,
+    MOCK_USER_BRIDGE_DATA,
+    MOCK_USER_VEDO_DATA,
+    VEDO_DEVICE_QUERY,
+)
+
+from tests.common import MockConfigEntry
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+from tests.typing import ClientSessionGenerator
+
+
+async def test_entry_diagnostics_bridge(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test Bridge config entry diagnostics."""
+    entry = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_BRIDGE_DATA)
+    entry.add_to_hass(hass)
+
+    with (
+        patch("aiocomelit.api.ComeliteSerialBridgeApi.login"),
+        patch(
+            "aiocomelit.api.ComeliteSerialBridgeApi.get_all_devices",
+            return_value=BRIDGE_DEVICE_QUERY,
+        ),
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert entry.state == ConfigEntryState.LOADED
+
+    entry_dict = entry.as_dict()
+    for key in TO_REDACT:
+        entry_dict["data"][key] = REDACTED
```

Comment:
Why do we have this? We want to test the code redacting itself right?

Reply:
Yup right, took from below link at my first attempt to write the test.

https://github.com/home-assistant/core/blob/1cc776d3327ff57ecb5ddba7f9764e099b2fd78f/tests/components/fritzbox/test_diagnostics.py#L27L29